packages/compiler-core/src/parse.ts

```diff
@@ -445,6 +445,7 @@ function parseTag(
 
     if (tag === 'slot') tagType = ElementTypes.SLOT
     else if (tag === 'template') tagType = ElementTypes.TEMPLATE
+    else if (tag === 'portal') tagType = ElementTypes.PORTAL
```

Comment:
Should also support `<Portal>`, so maybe use a regex here.

Reply:
If `portal` is equivalent citizen as `template` and `slot`, then a regexp isn't necessary, because `Template` and `Slot` aren't be treat as `template` and `slot`.

And `Portal` can be left for third-party libs, e.g., `PortalVue`, to use.

Reply:
`template` and `slot` are both native elements re-appropriated for Vue template syntax. `Portal`, `Suspense`, `Transition` and `KeepAlive` are built-in features that work conceptually like components, so they should support uppercase letter usage.

Reply:
Thanks to point out the difference. What about use built-in component map?

Specifically, we let `TransformContext` accepts a built-in component map, and in `transformElement`, we check if node's tag (lower cases) is built-in and map it to built-in name. For camelized  and pascalized names, we can registered the camelized built-in names in app's components config.

```js
// compiler-core
function transformElement(...) {
  ...
  // portal -> Symbol(Portal), transition -> Transition ?
  const builtinTag = context.builtins.get(node.tag)
  const args: CallExpression['arguments'] = [
      isComponent
        ? toValidAssetId(node.tag, `component`)
        : builtinTag || `"${node.tag}"`
  ...
}

```

Reply:
They should all be handled compile-time to avoid the runtime resolution cost.